src/post/post.service.ts

```diff
@@ -582,23 +591,23 @@ export class PostService {
 
         // If the provided handle is invalid, return early
         if (!isHandle(handle)) {
-            throw new Error('Invalid handle');
+            return error('invalid-handle');
         }
 
         if (!defaultAccount || !defaultAccount.id) {
-            throw new Error('Default account not found');
+            return error('missing-default-account');
         }
 
         // If the next parameter is not a valid URI, return early
         if (next !== '' && !isUri(next)) {
-            throw new Error('Invalid next parameter');
+            return error('invalid-next-parameter');
```

Comment:
Can we fix this at the type level to require passing a `URL` instead of a `string`?

Reply:
next can be empty string(`''`) in case when there is no next page, and we handle it differently. I am gonna leave it like this for now and try to come up with a cleaner approach while refactoring it to viewer.